### PR TITLE
[Keyboard] SKErgo

### PR DESCRIPTION
**Pull request by C1intMason**
## Description

Add SKErgo support

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Add SKErgo support

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/skergo/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+	[2] = LAYOUT_default(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
```
Comment (drashna) [inline keyboards/skergo/keymaps/default/keymap.c:24]:
```suggestion
		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS)

```

File: keyboards/skergo/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#include "skergo.h"
```
Comment (drashna) [inline keyboards/skergo/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/skergo/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "skergo.h"
```
Comment (drashna) [inline keyboards/skergo/keymaps/via/keymap.c:17]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/skergo/keymaps/via/rules.mk [hunk 4]
```diff
@@ -0,0 +1,4 @@
+VIA_ENABLE = yes
+LTO_ENABLE = yes
+EXTRAKEY_ENABLE = no
+MOUSEKEY_ENABLE = no
```
Comment (drashna) [inline keyboards/skergo/keymaps/via/rules.mk:4]: You probably want these enabled.

File: keyboards/skergo/rules.mk [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+# F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=2048
```
Comment (drashna) [inline keyboards/skergo/rules.mk:41]:
```suggestion
# Processor frequency.
F_CPU = 16000000

# Bootloader
BOOTLOADER = USBasp
```

File: keyboards/skergo/rules.mk [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+BOOTMAGIC_ENABLE ?= lite	# Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE ?= yes	# Mouse keys(+4700)
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= no		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
```
Comment (drashna) [inline keyboards/skergo/rules.mk:56]:
```suggestion
BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes	# Mouse keys
EXTRAKEY_ENABLE = yes	# Audio control and System control
CONSOLE_ENABLE = no	# Console for debug
COMMAND_ENABLE = no    # Commands for debug and configuration
SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
NKRO_ENABLE = no		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
AUDIO_ENABLE = no
RGBLIGHT_ENABLE = no
```

File: keyboards/skergo/skergo.h [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef KB_H
+#define KB_H
```
Comment (drashna) [inline keyboards/skergo/skergo.h:18]:
```suggestion
#pragma once
```

File: keyboards/skergo/skergo.h [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+	K400,       K402,       K404, K405,       K407,       K409,             K412, K413, K414  \
+) { \
+	{ K000,  K001,  K002,  K003,  K004,  K005,  K006,  K007,  K008,  K009,  K010,  K011,  K012,  K013,  K014 }, \
+	{ K100,  K101,  K102,  K103,  K104,  K105,  K106,  K107,  K108,  K109,  K110,  K111,  K112,  K113,  K114 }, \
+	{ K200,  K201,  K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  K210,  K211,  K212,  KC_NO, K214 }, \
+	{ K300,  K301,  K302,  K303,  K304,  K305,  K306,  K307,  K308,  K309,  K310,  K311,  KC_NO, K313,  K314 }, \
+	{ K400,  KC_NO, K402,  KC_NO, K404,  K405,  KC_NO, K407,  KC_NO, K409,  KC_NO, KC_NO, K412,  K413,  K414 }  \
+}
+
+#endif
```
Comment (drashna) [inline keyboards/skergo/skergo.h:36]:
```suggestion
```

File: keyboards/skergo/config.h [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_H
+#define CONFIG_H
```
Comment (drashna) [inline keyboards/skergo/config.h:18]:
```suggestion
#pragma once
```

File: keyboards/skergo/config.h [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4B49
+#define PRODUCT_ID      0x534B
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Keyzio
+#define PRODUCT         SKErgo
+#define DESCRIPTION     Through-hole Ergonomic Keyboard
```
Comment (drashna) [inline keyboards/skergo/config.h:28]:
```suggestion
```

File: keyboards/skergo/config.h [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+// #ifdef BACKLIGHT_PIN
+// #define BACKLIGHT_LEVELS 0
+// #endif
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/skergo/config.h:49]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/skergo/config.h [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+
+// #ifdef RGB_DI_PIN
+// #define RGBLIGHT_ANIMATIONS
+// #define RGBLED_NUM 0
+// #define RGBLIGHT_HUE_STEP 8
+// #define RGBLIGHT_SAT_STEP 8
+// #define RGBLIGHT_VAL_STEP 8
+// #endif
+
+#endif
```
Comment (drashna) [inline keyboards/skergo/config.h:74]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by C1intMason — COMMENTED** [review]
Suggestions lgtm, implemented everything.

**Review by drashna — APPROVED** [review]

File: keyboards/skergo/rules.mk [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no	# Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = no		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
```
Comment (fauxpark) [inline keyboards/skergo/rules.mk:23]:
```suggestion
# Processor frequency
F_CPU = 16000000

# Bootloader selection
BOOTLOADER = USBasp

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/skergo/info.json [hunk 14]
```diff
@@ -0,0 +1,86 @@
+{
+    "keyboard_name": "SKErgo",
+    "url": "https://keyz.io",
+    "maintainer": "Keyz.io",
+    "width": 19.25,
+    "height": 6,
+    "key_count": 67,
```
Comment (fauxpark) [inline keyboards/skergo/info.json:7]:
```suggestion
```

File: keyboards/skergo/skergo.h [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_default( \
```
Comment (fauxpark) [inline keyboards/skergo/skergo.h:21]:
```suggestion
#define LAYOUT( \
```

File: keyboards/skergo/readme.md [hunk 16]
```diff
@@ -0,0 +1,15 @@
+# SKErgo
+
+![skergo](https://cdn.shopify.com/s/files/1/0396/3001/2582/products/Keyzio-5_1024x1024@2x.jpg?v=1612909648)
```
Comment (fauxpark) [inline keyboards/skergo/readme.md:3]:
```suggestion
![skergo](https://i.imgur.com/WIqxiMfh.png)
```

File: keyboards/skergo/info.json [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+    "layouts": {
+      "LAYOUT_default": {
```
Comment (fauxpark) [inline keyboards/skergo/info.json:9]:
```suggestion
      "LAYOUT": {
```

File: keyboards/skergo/keymaps/default/keymap.c [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT_default(
```
Comment (fauxpark) [inline keyboards/skergo/keymaps/default/keymap.c:21]:
```suggestion
	[0] = LAYOUT(
```

File: keyboards/skergo/keymaps/default/keymap.c [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+		KC_ESC,  KC_1,    KC_2,   KC_3,  KC_4,   KC_5,    KC_6,    KC_7,    KC_8,    KC_9, KC_0, KC_MINS, KC_EQL, KC_GRV, KC_BSPC,
+		KC_TAB,  KC_Q,    KC_W,   KC_E,  KC_R,   KC_T,    KC_Y,    KC_U,    KC_I,    KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS, KC_PGUP,
+		KC_CAPS, KC_A,    KC_S,   KC_D,  KC_F,   KC_G,    KC_H,    KC_J,    KC_K,    KC_L, KC_SCLN, KC_QUOT, KC_ENT, KC_PGDN,
+		KC_LSFT, KC_Z,    KC_X,   KC_C,  KC_V,   KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT, KC_SLSH, KC_LSFT, KC_UP, KC_END,
+		KC_LCTL, KC_LALT, KC_SPC, MO(1), KC_SPC, KC_LCTL, KC_LEFT, KC_DOWN, KC_RGHT),
+
+	[1] = LAYOUT_default(
```
Comment (fauxpark) [inline keyboards/skergo/keymaps/default/keymap.c:28]:
```suggestion
	[1] = LAYOUT(
```

File: keyboards/skergo/keymaps/via/keymap.c [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] =	LAYOUT_default(
```
Comment (fauxpark) [inline keyboards/skergo/keymaps/via/keymap.c:21]:
```suggestion
	[0] =	LAYOUT(
```

File: keyboards/skergo/keymaps/via/keymap.c [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+			KC_ESC,  KC_1,    KC_2,   KC_3,    KC_4,   KC_5,    KC_6,    KC_7,    KC_8,    KC_9, KC_0,    KC_MINS, KC_EQL,  KC_GRV,  KC_BSPC,
+			KC_TAB,  KC_Q,    KC_W,   KC_E,    KC_R,   KC_T,    KC_Y,    KC_U,    KC_I,    KC_O, KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_PGUP,
+			KC_CAPS, KC_A,    KC_S,   KC_D,    KC_F,   KC_G,    KC_H,    KC_J,    KC_K,    KC_L, KC_SCLN, KC_QUOT, KC_ENT,  KC_PGDN,
+			KC_LSFT, KC_Z,    KC_X,   KC_C,    KC_V,   KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT, KC_SLSH, KC_LSFT, KC_UP, KC_END,
+			KC_LCTL, KC_LALT, KC_SPC, MO(1), KC_SPC, KC_LCTL, KC_LEFT, KC_DOWN, KC_RGHT),
+
+		[1] =	LAYOUT_default(
```
Comment (fauxpark) [inline keyboards/skergo/keymaps/via/keymap.c:28]:
```suggestion
		[1] =	LAYOUT(
```

File: keyboards/skergo/keymaps/via/keymap.c [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+		[2] =	LAYOUT_default(
```
Comment (fauxpark) [inline keyboards/skergo/keymaps/via/keymap.c:35]:
```suggestion
		[2] =	LAYOUT(
```

File: keyboards/skergo/keymaps/via/keymap.c [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+		[3] =	LAYOUT_default(
```
Comment (fauxpark) [inline keyboards/skergo/keymaps/via/keymap.c:42]:
```suggestion
		[3] =	LAYOUT(
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by C1intMason — COMMENTED** [review]
Commited changes

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Comment by C1intMason** [comment]
Thanks for the help!

**Pull request merged**